src/vim-loader/vim.ts

```diff
@@ -15,18 +15,17 @@ import { Object } from './object'
 export class Vim {
   source: string | undefined
   readonly document: IDocument
-  index: number = -1
-  settings: VimSettings
+  settings: VimConfig
```

Comment:
Minor - "settings" field could be renamed to "config" to keep the nomenclature consistent with VimConfig